### DISCUSSION TITLE
Type error: Object literal may only specify known properties, and 'username' does not exist in type 'UserWhereUniqueInput'

### DISCUSSION THREAD

**Discussion by shahbaz4783** [prisma/prisma#23208] [Q&A] [answered]
### Question

Im using prisma with nextjs.
It is working fine with local development, but when i deploy it on vercel, it throws the error:

```

Type error: Object literal may only specify known properties, and 'username' does not exist in type 'UserWhereUniqueInput'.
--
00:29:14.065 |  
00:29:14.065 | 29 \| 		// Check if username already exists
00:29:14.065 | 30 \| 		const usernameExists = await prisma.user.findUnique({
00:29:14.065 | > 31 \| 			where: { username: username },
00:29:14.065 | \| 			         ^
00:29:14.065 | 32 \| 		});
00:29:14.065 | 33 \| 		if (usernameExists) {
00:29:14.066 | 34 \| 			return NextResponse.json(
00:29:14.152 | Error: Command "npm run build" exited with 1


```

### Expected behavior (optional)

It should be deployed on vercel

### Information about Prisma Schema, Client Queries and Environment (optional)

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url = env("POSTGRES_PRISMA_URL")
  directUrl = env("POSTGRES_URL_NON_POOLING")
}

model User {
  id            String    @default(cuid()) @id
  name          String
  username      String    @unique
  email         String    @unique
  password      String
  createdAt     DateTime  @default(now()) @map(name: "created_at")
  updatedAt     DateTime  @updatedAt @map(name: "updated_at")
}

**Comment by nurul3101** [comment, marked as answer]
Hello @shahbaz4783 👋 

Is `npx prisma generate` a part of your build script when deploying to Vercel?
I believe the issue could be caused because your PrismaClient is outdated.
Here's an example for reference:

https://github.com/prisma/deployment-example-vercel/blob/main/package.json#L13

**Comment by shahbaz4783** [comment]
It worked after modifying the package.json!! Thanks for the help.

But in middle of deployment process, an error logged.
Attaching the screenshot.

<img width="794" alt="Screenshot 2024-02-22 at 11 45 29 AM" src="https://github.com/prisma/prisma/assets/96674638/16b46837-7cac-430c-a18a-c65122da539e">

**Comment by JamesKDouglas** [comment]
I'm facing this error as well, using Prisma and Next.js. The text of the function is,
"export async function updateSchedule(id: number, newScheduleData: Schedule){
    noStore();
    const schedule = await prisma.schedule.update({
        where: { id: id },
        data: {newScheduleData},
    })

    console.log("schedule updated!", schedule);
}"

Schedule is a type imported from the Prisma client. The error points to the line starting with "data...". I don't understand right now why the object gets provided to the function as an argument, no complaint is made, then there is a problem when it is provided to the Prisma client again.
